app/Http/Controllers/Device/Tabs/VlansController.php

```diff
@@ -51,6 +53,41 @@ public function name(): string
 
     public function data(Device $device): array
     {
-        return [];
+        return [
+            'vlans' => self::getVlans($device),
+            'submenu' => [
+                [
+                    ['name' => 'Basic', 'url' => ''],
+                ],
+                'Graphs' => [
+                    ['name' => 'Bits', 'url' => 'bits'],
+                    ['name' => 'Unicast Packets', 'url' => 'upkts'],
+                    ['name' => 'Non-Unicast Packets', 'url' => 'nupkts'],
+                    ['name' => 'Errors', 'url' => 'errors'],
+                ]
+            ],
+        ];
+    }
+
+    private function getVlans(Device $device)
+    {
+        // port.device needed to prevent loading device multiple times
+        $portVlan = PortVlan::where('device_id', $device->device_id)->with(['port.device', 'vlan1:vlan_vlan,vlan_name'])->has('port')->get();
+
+        foreach($portVlan as $pv) {
+            // TODO: Needed?
+            if (! $pv->untagged) {
+                if($pv->untagged == $pv->port->ifvlan) {
+                    $pv->untagged = 1;
+                }
+            }
```

Comment:
Is this needed?

Reply:
Different devices report untagged different ways.  I think this is needed.